### PR TITLE
Replace instructions to modify pubspec.yaml with `flutter pub add` command in multiple pages

### PR DESCRIPTION
**Pull request by miquelbeltran**
Following the proposal in #3340

I have replaced many cases in which the documentation asked developers to modify the pubspec.yaml to add a dependency, with a call to `flutter pub add`. This also required modifying a bit the description text.

## Presubmit checklist

- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

**Review by sfshaza2 — APPROVED** [review]
lgtm. Thanks, @miquelbeltran!

File: src/cookbook/navigation/set-up-universal-links.md [hunk 1]
```diff
@@ -39,14 +39,10 @@ It provides a simple API to handle complex routing scenarios.
...
 
-2. To include `go_router` package in your app, add a dependency for
-`go_router` in the `pubspec.yaml` file.
-
-    ```yaml
-    dependencies:
-      flutter:
-        sdk: flutter
-      go_router: ^6.0.9
+2. To include `go_router` package in your app, run `flutter pub add`:
```
Comment (parlough) [inline src/cookbook/navigation/set-up-universal-links.md:42]:
```suggestion
2. To include the `go_router` package as a dependency, 
   run `flutter pub add`:
```

File: src/cookbook/persistence/key-value.md [hunk 2]
```diff
@@ -37,14 +37,13 @@ This recipe uses the following steps:
...
 
-Before starting, add the [`shared_preferences`][]
-plugin to the `pubspec.yaml` file:
-
-```yaml
-dependencies:
-  flutter:
-    sdk: flutter
-  shared_preferences: "<newest version>"
+Before starting, add the [`shared_preferences`][] dependecy to your project.
```
Comment (parlough) [inline src/cookbook/persistence/key-value.md:40]:
```suggestion
Before starting, add the [`shared_preferences`][] package as a dependency.
```

File: src/cookbook/testing/unit/mocking.md [hunk 3]
```diff
@@ -54,14 +54,10 @@ so define that dependency in the `dependencies` section.
...
 
-```yaml
-dependencies:
-  http: <newest_version>
-dev_dependencies:
-  flutter_test:
-    sdk: flutter
-  mockito: <newest_version>
-  build_runner: <newest_version>
+To add the packages, run `flutter pub add`:
```
Comment (parlough) [inline src/cookbook/testing/unit/mocking.md:57]:
```suggestion
To add the dependency, run `flutter pub add`:
```

File: src/data-and-backend/json.md [hunk 4]
```diff
@@ -251,20 +251,10 @@ dependency, and two _dev dependencies_. In short, _dev dependencies_
 are dependencies that are not included in our app source code&mdash;they
 are only used in the development environment.
 
-The latest versions of these required dependencies can be seen by
-following the [pubspec file][] in the JSON serializable example.
+To add the packages run `flutter pub add`:
```
Comment (parlough) [inline src/data-and-backend/json.md:254]:
```suggestion
To add the dependencies, run `flutter pub add`:
```

**Review by parlough — APPROVED** [review]
Thanks so much for this!!

Some minor suggestions to consider:

File: src/cookbook/testing/unit/mocking.md (hunk 3 above)
Comment (miquelbeltran) [reply src/cookbook/testing/unit/mocking.md:57]: Note that here it is referring to multiple dependencies (mockito, build_runner, etc.) and not a single one

Comment (parlough) [reply src/cookbook/testing/unit/mocking.md:57]: Oops, sorry. How about:

```suggestion
To add the dependencies, run `flutter pub add`:
```

**Pull request merged**